### ISSUE TITLE
build: replace cmake by bazel.

### ISSUE THREAD

**Issue by LittleFall** [pingcap/tiflash#6266]
## Enhancement

currently, our cmake-based build system is too complicated to understand and modify, maybe we can replace it by [Bazel](https://bazel.build/).

for example, grpc's bazel config: https://github.com/grpc/grpc/blob/7897ae9308a25a064ee2e7386da25d37cc3273fa/BUILD

**Comment by JaySon-Huang** [comment]
> cmake-based build system is too complicated

The 1544 lines in bazel config is more complicated than CMake to me...

**Comment by Lloyd-Pottiger** [comment]
> > cmake-based build system is too complicated
> 
> The 1544 lines in bazel config is more complicated than CMake to me...

Agree, the prons of bazel are fast and deterministic incremental builds rather than simple. Another difficulty is that it is hard for bazel to integrate third-party repo.

**Comment by fuzhe1989** [comment]
How about [xmake](https://xmake.io/):
- compatible with existed cmake scripts
- dependency package integration
- fast and readable

**Comment by breezewish** [comment]
There are a lot of existing discussions about Bazel and cmake, may be helpful:

https://www.reddit.com/r/cpp/comments/vxr1ap/bazel_or_cmake/
